src/types/profile.js

```diff
@@ -240,29 +254,68 @@ export type ExtensionTable = {|
  * Meta information associated for the entire profile.
  */
 export type ProfileMeta = {|
-  interval: number,
+  // The interval at which the threads are sampled.
+  interval: Milliseconds,
+  // The number of milliseconds since midnight January 1, 1970 GMT.
   startTime: Milliseconds,
-  abi: string,
-  misc: string,
-  oscpu: string,
-  platform: string,
-  processType: number, // TODO find the possible values
+  // The process type where the Gecko profiler was started. This is the raw enum
+  // numeric value as defined here:
+  // https://searchfox.org/mozilla-central/rev/819cd31a93fd50b7167979607371878c4d6f18e8/xpcom/build/nsXULAppAPI.h#365
+  processType: number,
   // The extensions property landed in Firefox 60, and is only optional because older
   // processed profile versions may not have it. No upgrader was written for this change.
   extensions?: ExtensionTable,
+  // The list of categories as provided by the platform.
   categories: CategoryList,
-  product: string,
-  stackwalk: number,
-  toolkit: string,
+  // The name of the product, most likely "Firefox".
+  product: 'Firefox' | string,
+  // This value represents a boolean, but for some reason is written out as an int value.
+  // It's 0 for the stack walking feature being turned off, and 1 for stackwalking being
+  // turned on.
+  stackwalk: 0 | 1,
+  // Before processing, this was the Gecko profile version.
```

Comment:
I clarified this a bit, since "Gecko profile format" is the technical name for the version:

```js
  // This is the Gecko profile format version (which is the unprocessed version received
  // directly from the browser.)
  version: number,
  // This is the processed profile format version.
  preprocessedProfileVersion: number,
```